src/node_crypto.cc

```diff
@@ -608,8 +584,74 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
   }
 
  end:
-  if (x != nullptr)
+  if (ret && x != nullptr) {
     *cert = x;
+    CRYPTO_add(&(*cert)->references, 1, CRYPTO_LOCK_X509);
```

Comment:
That's a lot of spooky action at a distance.  This function is not even a member of that class.